src/Sets/Zonotope.jl

```diff
@@ -387,3 +364,143 @@ function split(Z::Zonotope, j::Int)
+const _GIR05 = GIR05()
+
+# return the indices of the generators in G (= columns) sorted according to the COMB03 method
+# the generator index with highest score goes first
+function _weighted_gens!(indices, G::AbstractMatrix{N}, ::COMB03) where {N}
+    p = size(G, 2)
+    weights = Vector{N}(undef, p)
+    @inbounds for j in 1:p
+        v = view(G, :, j)
+        weights[j] = norm(v, 2)
+    end
+    sortperm!(indices, weights, rev=true, initialized=false)
+    return indices
+end
+
+# return the indices of the generators in G (= columns) sorted according to the GIR05 method
+# the generator index with highest score goes first
+function _weighted_gens!(indices, G::AbstractMatrix{N}, ::GIR05) where {N}
+    n, p = size(G)
+    weights = Vector{N}(undef, p)
+    @inbounds for j in 1:p
+        aux_norm_1 = zero(N)
+        aux_norm_inf = zero(N)
+        for i in 1:n
+            abs_Gij = abs(G[i, j])
+            aux_norm_1 += abs_Gij
+            if aux_norm_inf < abs_Gij
+                aux_norm_inf = abs_Gij
+            end
+        end
+        weights[j] = aux_norm_1 - aux_norm_inf
+    end
+    sortperm!(indices, weights, rev=true, initialized=false)
+    return indices
+end
+
+# compute interval hull of the generators of G (= columns) corresponding to `indices`
+function _interval_hull(G::AbstractMatrix{N}, indices) where {N}
+    n, p = size(G)
+    Lred = zeros(N, n, n)
+    @inbounds for i in 1:n
+        for j in indices
+            Lred[i, i] += abs(G[i, j])
+        end
+    end
+    return Lred
+end
+
+# implementation for static arrays
+function _interval_hull(G::SMatrix{n, p, N, L}, indices) where {n, p, N, L}
+    Lred = zeros(MMatrix{n, n, N})
+    @inbounds for i in 1:n
+        for j in indices
+            Lred[i, i] += abs(G[i, j])
+        end
+    end
+    return SMatrix{n, n}(Lred)
+end
+
+# given an n x p matrix G and a vector of m integer indices with m <= p,
+# concatenate the columns of G given by `indices` with the matrix Lred
+function _hcat_KLred(G::AbstractMatrix, indices, Lred::AbstractMatrix)
+    K = view(G, :, indices)
+    return hcat(K, Lred)
+end
+
+# implementation for static arrays
+function _hcat_KLred(G::SMatrix{n, p, N, L1}, indices, Lred::SMatrix{n, n, N, L2}) where {n, p, N, L1, L2}
+    m = length(indices)
+    K = SMatrix{n, m}(view(G, :, indices))
+    return hcat(K, Lred)
+end
+
+# Implements zonotope order reduction method from [COMB03]
+# We follow the notation from [YS18]
+function _reduce_order(Z::Zonotope, r::Number, ::COMB03)
+    r >= 1 || throw(ArgumentError("the target order should be at least 1, but it is $r"))
+    c = Z.center
+    G = Z.generators
+    n, p = size(G)
+
+    # r is bigger than the order of Z => don't reduce
+    (r * n >= p) && return Z
+
+    # this algorithm sort generators by decreasing 2-norm
+    indices = Vector{Int}(undef, p)
+    _weighted_gens!(indices, G, _COMB03)
+
+    # the first m generators have greatest weight
+    m = floor(Int, n * (r - 1))
```

Comment:
in fact if `r = 1` => `m = 0` and the generators dont need to be sorted

Reply:
I double checked and this is indeed correct, giving a factor x5 .. x10 speedup for such case.

Reply:
This is a rather rare case because you would probably use `box_approximation` instead, but okay :+1:

Reply:
That's true, though it's convenient to rely on `reduce_order` being fast in other place (such as [here](https://github.com/JuliaReach/ReachabilityAnalysis.jl/blob/3c3d4a62b514d24773d3dee53b798e0200b4e1bf/src/Algorithms/GLGM06/post.jl#L39)). 

BTW after merging this PR I'll remove a bunch of repeated code in RA and tag releases on both repos.